src/os/windows/win32.cpp

```diff
@@ -544,10 +544,10 @@ std::string FS2OTTD(const std::wstring &name)
 	int name_len = (name.length() >= INT_MAX) ? INT_MAX : (int)name.length();
 	int len = WideCharToMultiByte(CP_UTF8, 0, name.c_str(), name_len, nullptr, 0, nullptr, nullptr);
 	if (len <= 0) return std::string();
-	char *utf8_buf = AllocaM(char, len + 1);
+	std::string utf8_buf(len + 1, '\0');
```

Comment:
```suggestion
	std::string utf8_buf(len, '\0');
```
WideCharToMultiByte() returned length includes terminating null character.

The old code used +1 for extra safety but cropped the result on return.